### PR TITLE
Demo2 issue fixes

### PR DESCRIPTION
**Pull request by rachana-egov**
## Release Notes

- **New Features**
	- Enhanced pagination options in multiple components, allowing users to select from `[10, 20, 50, 100]` rows per page.
	- Added toast notifications to provide user feedback on actions in the `PopInbox` component.

- **Improvements**
	- Streamlined the `WorkflowCommentPopUp` by integrating button functionalities directly within the popup.
	- Improved accessibility and security components to track changes and manage save button states effectively.

- **Bug Fixes**
	- Corrected state management for various components to ensure accurate data handling and user interactions.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/securityPopUp.js [hunk 1]
```diff
@@ -12,11 +12,15 @@ const SecurityPopUp = ({ onClose, census, onSuccess }) => {
 
   // Initialize answers with the existing security details from additionalDetails if available
   const [answers, setAnswers] = useState({});
+  const [initialAnswers, setInitialAnswers] = useState({});
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/securityPopUp.js:15]: it is already there

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/securityPopUp.js [hunk 2]
```diff
@@ -45,6 +48,13 @@ const SecurityPopUp = ({ onClose, census, onSuccess }) => {
     },
   };
 
+  // Check if the answers have changed from the initial values
+  const isChanged = () => {
+    return Object.keys(answers).some(
+      (key) => answers[key] !== initialAnswers[key]
+    );
+  };
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/securityPopUp.js:56]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/accessbilityPopUP.js [hunk 3]
```diff
@@ -60,6 +69,15 @@ const AccessibilityPopUp = ({ onClose, census, onSuccess }) => {
...
   };
 
+  // Check if dropdown values have changed from the initial values
+  const isChanged = () => {
+    return (
+      dropdown1Value !== initialValues.roadCondition ||
+      dropdown2Value !== initialValues.terrain ||
+      dropdown3Value !== initialValues.transportationMode
+    );
+  };
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/accessbilityPopUP.js:79]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/accessbilityPopUP.js [hunk 4]
```diff
@@ -12,13 +12,22 @@ const AccessibilityPopUp = ({ onClose, census, onSuccess }) => {
...
+      setDropdown1Value(roadCondition);
+      setDropdown2Value(terrain);
+      setDropdown3Value(transportationMode);
+
+      // Store initial values to compare later
+      setInitialValues({
+        roadCondition,
+        terrain,
+        transportationMode,
+      });
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/accessbilityPopUP.js:30]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/PopInbox.js [hunk 5]
```diff
@@ -219,38 +220,35 @@ const PopInbox = () => {
...
-    setLimitAndOffset({ ...limitAndOffset, offset: (page - 1) * 5 })
+    setLimitAndOffset({ ...limitAndOffset, offset: (page - 1) * rowsPerPage })
   }
 
   const handlePerRowsChange = (currentRowsPerPage, currentPage) => {
     setRowsPerPage(currentRowsPerPage);
-    setCurrentPage(currentPage);
-    setLimitAndOffset({ limit: currentRowsPerPage, offset: (currentPage - 1) * currentRowsPerPage })
+    setCurrentPage(1);
+    setLimitAndOffset({ limit: currentRowsPerPage, offset: (currentPage - 1) * rowsPerPage })
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/PopInbox.js:251]: it is initially set in same way

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**